InstanceManager.js

```diff
@@ -647,97 +730,88 @@ class InstanceManager {
-          if (instance.status == 'EXITED') {
-            if (process.env.LOG_IMMEDIATE && process.env.LOG_IMMEDIATE == "1") {
-              console.log((new Date()).toISOString()
-                + " innerCheckDown resolve for " + instance.name);
-            }
-            resolve(instance);
-          } else if (++attempts === killAttempts) {
-            if (process.env.LOG_IMMEDIATE && process.env.LOG_IMMEDIATE == "1") {
-              console.log((new Date()).toISOString()
-                + " innerCheckDown: killed" + instance.name);
-            }
-            instance.process.kill('SIGKILL');
-            instance.status = 'KILLED';
-            setTimeout(innerCheckDown, waitInterval);
-          } else if (attempts > maxAttempts) {
-            if (process.env.LOG_IMMEDIATE && process.env.LOG_IMMEDIATE == "1") {
-              console.log((new Date()).toISOString()
-                + " innerCheckDown reject for " + instance.name);
-            }
-            reject(new Error(instance.name + ' did not stop gracefully after ' + (waitInterval * attempts) + 'ms'));
-          } else {
-            setTimeout(innerCheckDown, waitInterval);
-          }
-        })();
-      });
+  async shutdown(instance) {
+    if (instance.status == 'EXITED') {
+      return instance;
     }
+    debugLog(`Shutdown ${instance.name}`);
 
-    return rp.delete({
-      url: this.getEndpointUrl(instance) + '/_admin/shutdown',
-    })
-    .catch(err => {
+    try {
+      await rp.delete({ url: this.getEndpointUrl(instance) + '/_admin/shutdown' })
+    } catch (err) {
+      let expected = false;
       if (err && (err.statusCode === 503 || err.error)) {
-        if (err.error.code == 'ECONNREFUSED') {
-          console.warn('hmmm...server ' + instance.name + ' did not respond (' + err.code + '). Assuming it is dead. Status is: ' + instance.status);
-          return checkDown();
-        } else if (err.error.code == 'ECONNRESET') {
-          return checkDown();
+        // Some "errors" are expected.
+        let errObj = (err.error instanceof String || typeof err.error === "string") ?
+                       JSON.parse(err.error) : err.error;
+        if (errObj.code == 'ECONNREFUSED') {
+          console.warn('hmmm...server ' + instance.name + ' did not respond (' + JSON.stringify(errObj) + '). Assuming it is dead. Status is: ' + instance.status);
+          expected = true;
+        } else if (errObj.code == 'ECONNRESET') {
+          expected = true;
         } else if (err.statusCode === 503) {
           console.warn('server ' + instance.name + ' answered 503. Assuming it is shutting down. Status is: ' + instance.status);
-          return checkDown();
+          expected = true;
+        } else if (errObj.errorNum === ERR_IN_SHUTDOWN) {
+          expected = true;
+        }
+        if (!expected) {
+          console.error("Wir hassen den shutdown: ", errObj.errorNum, " so total sehr " , err.error, "wir haben den typ", typeof err.error);
```

Comment:
English please. I don't mind flavor text but let's not obscure the meaning for our non-German speaking devs.